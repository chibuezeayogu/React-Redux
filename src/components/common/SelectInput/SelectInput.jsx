import React from 'react';
import {
	string, func, arrayOf, object 
} from 'prop-types';

const SelectInput = ({
	name, label, onChange, defaultOption, value, error, options 
}) => (
	<div className="form-group">
		<label htmlFor={name}>{label}</label>
		<div className="field">
			<select
				name={name}
				value={value}
				onChange={onChange}
				className="form-control">
				<option value="">{defaultOption}</option>
				{options.map(option => (
					<option key={option.value} value={option.value}>{option.text}</option>
				))}
			</select>
			{error && <div className="alert alert-danger">{error}</div>}
		</div>
	</div>
);

SelectInput.propTypes = {
	name: string.isRequired,
	label: string.isRequired,
	onChange: func.isRequired,
	defaultOption: string,
	value: string,
	error: string,
	options: arrayOf(object)
};

export default SelectInput;
