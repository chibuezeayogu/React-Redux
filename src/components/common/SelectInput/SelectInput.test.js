import React from 'react';
import { shallow } from 'enzyme';

import SelectInput from './SelectInput';

describe('SelectInput Component', () => {
	const props = {
		name: '',
		label: '',
		onChange: jest.fn(),
		defaultOption: '',
		value: '',
		error: 'error',
		options: [{ value: 'text', text: 'text' }]
	};

	const wrapper = shallow(<SelectInput {...props} />);
	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
