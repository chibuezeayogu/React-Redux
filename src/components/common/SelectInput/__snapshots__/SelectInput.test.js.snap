// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput Component should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor=""
  >
    <span>
      error
    </span>
  </label>
  <div
    className="field"
  >
    <select
      className="form-control"
      name=""
      onChange={[MockFunction]}
      value=""
    >
      <option
        value=""
      />
      <option
        key="text"
        value="text"
      >
        text
      </option>
    </select>
  </div>
</div>
`;
